opensearch-operator/controllers/dashboard_test.go

```diff
@@ -44,6 +46,44 @@ var _ = Describe("Dashboards Reconciler", func() {
 			}, timeout, interval).Should(BeTrue())
 		})
 
+		It("should create the secret for volumes", func() {
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-secret",
+					Namespace: OpensearchCluster.Namespace,
+				},
+				StringData: map[string]string{
+					"test.yml": "foobar",
+				},
+			}
+			Expect(k8sClient.Create(context.Background(), secret)).To(Succeed())
+			Eventually(func() error {
+				return k8sClient.Get(context.Background(), types.NamespacedName{
+					Name:      secret.Name,
+					Namespace: secret.Namespace,
+				}, &corev1.Secret{})
+			}, timeout, interval).Should(Succeed())
+		})
+
+		It("should create the configmap for volumes", func() {
+			cm := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cm",
+					Namespace: OpensearchCluster.Namespace,
+				},
+				Data: map[string]string{
+					"test.yml": "foobar",
+				},
+			}
+			Expect(k8sClient.Create(context.Background(), cm)).To(Succeed())
+			Eventually(func() error {
+				return k8sClient.Get(context.Background(), types.NamespacedName{
+					Name:      cm.Name,
+					Namespace: cm.Namespace,
+				}, &corev1.ConfigMap{})
+			}, timeout, interval).Should(Succeed())
+		})
+
```

Comment:
Missing a test to see if the secret/configmap are actually mounted to the dashboards deployment